templates/hoover-deps.nomad

```diff
@@ -20,26 +20,126 @@ job "hoover-deps" {
           "{% raw %}${meta.liquid_volumes}{% endraw %}/hoover/es/data:/usr/share/elasticsearch/data",
         ]
         port_map {
-          es = 9200
+          http = 9200
+          transport = 9300
         }
         labels {
           liquid_task = "hoover-es"
         }
       }
       env {
         cluster.name = "hoover"
-        ES_JAVA_OPTS = "-Xms${config.elasticsearch_heap_size}m -Xmx${config.elasticsearch_heap_size}m -XX:+PrintGCDetails -XX:+UnlockDiagnosticVMOptions"
+        node.name = "master"
+
+        http.port = "9200"
+        http.publish_port = "{% raw %}${NOMAD_HOST_PORT_http}{% endraw %}"
+        http.bind_host = "0.0.0.0"
+        http.publish_host = "{% raw %}${attr.unique.network.ip-address}{% endraw %}"
+
+        transport.port = "9300"
+        transport.publish_port = "{% raw %}${NOMAD_HOST_PORT_transport}{% endraw %}"
+        transport.bind_host = "0.0.0.0"
+        transport.publish_host = "{% raw %}${attr.unique.network.ip-address}{% endraw %}"
+
+        ES_JAVA_OPTS = "-Xms${config.elasticsearch_heap_size}m -Xmx${config.elasticsearch_heap_size}m -XX:+UnlockDiagnosticVMOptions"
+      }
+      resources {
+        memory = ${config.elasticsearch_memory_limit}
+        network {
+          mbits = 1
+          port "http" {}
+          port "transport" {}
+        }
+      }
+      service {
+        name = "hoover-es-master"
+        port = "http"
+        tags = ["snoop-/_es strip=/_es"]
+        check {
+          name = "http"
+          initial_status = "critical"
+          type = "http"
+          path = "/_cluster/health"
+          interval = "${check_interval}"
+          timeout = "${check_timeout}"
+        }
+      }
+      service {
+        name = "hoover-es-master-transport"
+        port = "transport"
+        check {
+          name = "transport"
+          initial_status = "critical"
+          type = "tcp"
+          interval = "${check_interval}"
+          timeout = "${check_timeout}"
+        }
+      }
+    }
+
+    ${ promtail_task() }
+  }
+
+  {% if config.elasticsearch_data_node_count %}
```

Comment:
So if `elasticsearch_data_node_count` is zero does elasticsearch still work? If so, then I'm pretty sure you can drop this `if` block, because `group { count = 0 }` is fine.

Reply:
We'd wait for a health check that doesn't exist, because we don't check the service count.
I think it's cleaner this way, since 0 is the default, most users won't even see this.